src/main/java/core/basesyntax/JavaStreamApi.java

```diff
@@ -3,88 +3,73 @@
 import java.util.Collections;
 import java.util.List;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 public class JavaStreamApi {
 
-    /**
-     * <p>1. Дано: List of Integer numbers.
-     * Вернуть сумму нечетных числел или 0, если таких несуществует</p>
-     **/
     public Integer oddSum(List<Integer> numbers) {
-        return null;
+        return numbers.stream()
+                .filter(number -> number % 2 == 1)
+                .reduce(0, (x, y) -> x + y);
     }
 
-    /**
-     * <p>2. Дана коллекция строк List of String elements
-     * (пример: Arrays.asList(«a1», «a2», «a3», «a1»)).
-     * Вернуть количество вхождений объекта `element`</p>
-     **/
     public Long elementCount(List<String> elements, String element) {
-        return null;
+        return elements.stream()
+                .filter(s -> s.equals(element))
+                .count();
     }
 
-    /**
-     * <p>3. Дана коллекция строк List of String elements
-     * (пример: Arrays.asList(«a1», «a2», «a3», «a1»)).
-     * Вернуть Optional первого элемента коллекции</p>
-     **/
     public Optional<String> firstElement(List<String> elements) {
-        return null;
+        return elements.stream()
+                .findFirst();
     }
 
-    /**
-     * <p>4. Дана коллекция строк List of String elements
-     * (приметр: Arrays.asList(«a1», «a2», «a3», «a1»)).
-     * Найти элемент в коллекции равный `element` или кинуть ошибку NoSuchElementException</p>
-     **/
     public String findElement(List<String> elements, String element) {
-        return null;
+        return elements.stream()
+                .filter(x -> x.equals(element))
+                .findFirst()
+                .orElseThrow();
     }
 
-    /**
-     * <p>5. Дана коллекция чисел List of Integer numbers (пример: Arrays.asList(6, 2, 3, 7, 2, 5))
-     * Отнимите от каждого элемента, который стоит на непарной позиции (имеет не парный индекс) 1
-     * и верните среднее арифметическое всех нечетных чисел или киньте ошибку
-     * NoSuchElementException</p>
-     **/
     public Double averageSumOdd(List<Integer> numbers) {
-        return null;
+        int[] index = {0};
+        return numbers.stream()
+                .mapToInt(x -> index[0]++ % 2 == 1 ? x - 1 : x)
```

Comment:
let's create stream of integers and use them as indexes for numbers

Reply:
done